opslevel/resource_opslevel_service_dependency.go

```diff
@@ -1,5 +1,209 @@
 package opslevel
 
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &ServiceDependencyResource{}
+
+var _ resource.ResourceWithImportState = &ServiceDependencyResource{}
+
+func NewServiceDependencyResource() resource.Resource {
+	return &ServiceDependencyResource{}
+}
+
+// ServiceDependencyResource defines the resource implementation.
+type ServiceDependencyResource struct {
+	CommonResourceClient
+}
+
+// ServiceDependencyResourceModel describes the ServiceDependency managed resource.
+type ServiceDependencyResourceModel struct {
+	DependsUpon types.String `tfsdk:"depends_upon"`
+	Id          types.String `tfsdk:"id"`
+	Note        types.String `tfsdk:"note"`
+	Service     types.String `tfsdk:"service"`
+}
+
+func NewServiceDependencyResourceModel(ctx context.Context, serviceDependency opslevel.ServiceDependency) ServiceDependencyResourceModel {
+	return ServiceDependencyResourceModel{
+		Id:   ComputedStringValue(string(serviceDependency.Id)),
+		Note: OptionalStringValue(serviceDependency.Notes),
+	}
+}
+
+func (r *ServiceDependencyResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_serviceDependency"
+}
+
+func (r *ServiceDependencyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "ServiceDependency Resource",
+
+		Attributes: map[string]schema.Attribute{
+			"depends_upon": schema.StringAttribute{
+				Description: "The ID or alias of the service that is depended upon.",
+				Optional:    true,
```

Comment:
This should be required?

Reply:
Totally, the old code shows Optional but that doesn't make sense